go/chat/teams.go

```diff
@@ -80,22 +80,20 @@ func (t *ImplicitTeamsNameInfoSource) Lookup(ctx context.Context, name string, v
 		return t.lookupInternalName(ctx, name, vis)
 	}
 
-	teamID, teamName, err := teams.LookupOrCreateImplicitTeam(ctx, t.G().ExternalG(), name, vis == keybase1.TLFVisibility_PUBLIC)
+	teamID, teamName, impTeamName, err := teams.LookupOrCreateImplicitTeam(ctx, t.G().ExternalG(), name,
+		vis == keybase1.TLFVisibility_PUBLIC)
 	if err != nil {
 		return res, err
 	}
-
 	if !teamID.IsRootTeam() {
 		panic(fmt.Sprintf("implicit team found via LookupImplicitTeam not root team: %s", teamID))
 	}
 
+	res.CanonicalName = teamName.String()
```

Comment:
It seems like letting the __ugly_name float around out here in chat is an unnecessary wart. It's annoying to have so many identifiers, and the __ugly_name only exists as a consequence of the fact that iteams happen to be built on teams. Could you use the team ID as the canonical name?

Sorry for stirring up so much ruckus on this PR.

Reply:
This name gets written into the TLF name field of messages, it isn't displayed to anyone. I assume the name of this team needs to be represented somewhere on the message.

Reply:
Why would the __ugly_name need to be represented somewhere? I feel like it should be avoided.

Reply:
Is there any substantive difference here? If we aren't going to display it anywhere, than what difference does it make if it is this name vs the ID? Is there something intrinsically nicer about an ID? At the bare minimum it would be nice to see the name when debugging, it may have a use. I already have the ID represented on the message.

Reply:
I think it's a footgun to have multiple IDs and ways of addressing objects that need to be checked converted. But no there's no substantive difference. Not sure why it's nice to see for debugging.